app/src/main/java/org/mozilla/focus/session/SessionNotificationService.kt

```diff
@@ -150,17 +151,24 @@ class SessionNotificationService : Service() {
             val intent = Intent(context, SessionNotificationService::class.java)
             intent.action = ACTION_START
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                context.startForegroundService(intent)
-            } else {
-                context.startService(intent)
-            }
+            // Post starting the service to the end of the main thread to give it the best
```

Comment:
nit: "to give it the best chance to complete in time."

In time? What issues do we see if we don't let it complete? Elaborate! This issue is serious/complex enough that'd I'd include an issue number.

You may also want to mention this is a speculative fix and that some processing still occurs afterwards.